src/enzo/Group_ReadDataHierarchy.C

```diff
@@ -225,6 +225,10 @@ int Group_ReadDataHierarchy(FILE *fptr, hid_t Hfile_id, HierarchyEntry *Grid,
                                      MetaData.PPMFlatteningParameter,
                                      MetaData.PPMDiffusionParameter,
                                      MetaData.PPMSteepeningParameter);
+  if (SelfGravity) {
```

Comment:
As noted in another comment, this could include subgrids by checking ParentGrid (to see if it is NULL).